app/src/main/java/com/example/househomey/tags/TagFragment.java

```diff
@@ -0,0 +1,83 @@
+package com.example.househomey.tags;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import androidx.fragment.app.DialogFragment;
+
+import com.example.househomey.R;
+import com.google.android.material.chip.Chip;
+import com.google.android.material.chip.ChipGroup;
+
+/**
+ * DialogFragment that allows users to add new tags
+ * @author Matthew Neufeld
+ */
+public class TagFragment extends DialogFragment {
+
+    // Declare UI components
+    private ChipGroup chipGroup;
+    private EditText tagEditText;
+    private Button addTagButton;
+
+    /**
+     * Called to create the dialog, initializing UI components and setting up button listeners.
+     * @param savedInstanceState Bundle containing the saved state of the fragment.
+     * @return The created AlertDialog.
+     */
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+
+        // Initialize AlertDialog builder
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+
+        // Inflate the layout for this fragment
+        LayoutInflater inflater = requireActivity().getLayoutInflater();
+        View rootView = inflater.inflate(R.layout.fragment_tags, null);
+
+        // Initialize UI components
+        chipGroup = rootView.findViewById(R.id.chip_group_labels);
+        tagEditText = rootView.findViewById(R.id.tag_edit_text);
+        addTagButton = rootView.findViewById(R.id.add_tag_button);
+
+        // Set up button click listener to add tags
+        addTagButton.setOnClickListener(v -> addTag());
+
+        // Set up the AlertDialog
+        builder.setView(rootView);
+        builder.setTitle("Tags");
+        builder.setNeutralButton("Cancel", null);
+        builder.setPositiveButton("Apply Tags", null);
+
+        return builder.create();
+    }
+
+    /**
+     * Adds a new tag to the ChipGroup when addTagButton is clicked.
+     */
+    private void addTag() {
+        String tagText = tagEditText.getText().toString().trim();
+
+        // Check if the tag is not empty
+        if (!tagText.isEmpty()) {
+            // Create a new Chip and set its properties
+            Chip chip = new Chip(requireContext());
```

Comment:
Not sure if its viable/feasible but could we consolidate the keyword chip and tag chip into some util method like ```setupChip(chipText, chipGroup);``` ? Look at autoFillFilter() in KeywordFilterFragment.

Reply:
@ldbonkowski See latest commit, I applied the util to TagFragment and KeywordFilterFragment

Reply:
Do we not want to duplicate the colour lines too? 
``` chip.setChipBackgroundColorResource(R.color.white);
chip.setChipStrokeColorResource(R.color.brown);
chip.setTextColor(ContextCompat.getColor(context , R.color.brown));```

Also I was thinking we could call it newChip and put line 68 inside as well and just pass the context in.

Reply:
We can't put line 68 in there because keyword and tag chips are configured differently. The same reason applies to why we can't add in the colour lines unless we want to do a bit of a revamp.

Reply:
@ldbonkowski Addressed these in the most recent commit. I think it's as consolidated as we can get for now. I do see areas that we could potentially consolidate in the future once we have selecting and applying tag functionality. But I will tackle that once we have that feature.